test/testdata/desugar/range.rb

```diff
@@ -8,15 +8,18 @@ def foo
     f = ('a'..'z').last
 
     T.reveal_type(a) # error: Revealed type: `T::Range[Integer]`
-    T.reveal_type(b) # error: Revealed type: `T::Range[Integer]`
+    T.reveal_type(b) # error: Revealed type: `T::Range[T.untyped]`
     T.reveal_type(c) # error: Revealed type: `T::Range[Integer]`
-    T.reveal_type(d) # error: Revealed type: `T::Range[Integer]`
+    T.reveal_type(d) # error: Revealed type: `T::Range[T.untyped]`
     T.reveal_type(e) # error: Revealed type: `Integer`
     T.reveal_type(f) # error: Revealed type: `String`
```

Comment:
Could you add a few cases that use `T::Range[...].new` as well?

Reply:
I completely rewrote the test file to be more exhaustive on the three forms: `x..y`, `Range.new` and `T::Range[X].new`. Let me know if you want to add more test cases.